app/bundles/CoreBundle/Assets/js/14.notifications.js

```diff
@@ -0,0 +1,19 @@
+Mautic.notificationIndexLoad = function (translations) { 
+    mQuery(document).ready(function() { 
+        // Hide the New button
+        mQuery("a[href$='account/notifications/new']").hide();
+
+        // Mark a notification as read.
+        mQuery('#notificationTable a')
+            .each(function(index) { 
+                mQuery(this).on('click', function() { 
+                    var tr = mQuery(this).parent().parent();
+                    var tdId = tr.children('#notificationId');
+                    var id = tdId.text(); 
+                    var children = tr.children('#isRead');
+                    children.text(translations['mautic.core.yes']);
+                    Mautic.clearNotification(id);
```

Comment:
There must be an easier way of doing this. All of this Dom navigation to mark notification as read en-masse seems like a waste of time.  

Why not just instead of ar children = tr.children('#isRead'); do mQuery('#isRead').text((translations['mautic.core.yes']))
?

Reply:
We need to get the notification ID to pass to `Mautic.clearNotification(id)`

Reply:
So? You can do that just fine. You can use HTML5 attributes and in the function get that from the element that triggers the event